src/libmongoc/src/mongoc/mongoc-trace-private.h

```diff
@@ -28,53 +28,82 @@
 #include <mongoc/mongoc-log-private.h>
 #include <mongoc/mongoc-config.h>
 
+#include <mlib/config.h>
+
 
 BSON_BEGIN_DECLS
 
 // `gLogTrace` determines if tracing is enabled at runtime.
 extern bool gLogTrace;
 
-#define TRACE(msg, ...)                        \
-   do {                                        \
-      if (MONGOC_TRACE_ENABLED && gLogTrace) { \
-         mongoc_log (MONGOC_LOG_LEVEL_TRACE,   \
-                     MONGOC_LOG_DOMAIN,        \
-                     "TRACE: %s():%d " msg,    \
-                     BSON_FUNC,                \
-                     (int) (__LINE__),         \
-                     __VA_ARGS__);             \
-      }                                        \
-   } while (0)
+#define TRACE(msg, ...)                                         \
+   if (1) {                                                     \
+      mlib_diagnostic_push ();                                  \
+      mlib_disable_constant_conditional_expression_warnings (); \
+      if (MONGOC_TRACE_ENABLED && gLogTrace) {                  \
```

Comment:
Confirming: does this trigger C4127? I expect the check with `gLogTrace` would make the conditional non-constant.

Reply:
Yes, given `MONGOC_TRACE_ENABLED = 0` (false), short-circuiting skips `gLogTrace`, thus the condition is "always false" at compile-time.

Reply:
Ah, right. Resolving.